src/pages/workspace/WorkspaceNewRoomPage.tsx

```diff
@@ -32,98 +31,68 @@ import * as Report from '@userActions/Report';
 
     /** Form state for NEW_ROOM_FORM */
-    formState: PropTypes.shape({
-        /** Loading state for the form */
-        isLoading: PropTypes.bool,
-
-        /** Field errors in the form */
-        errorFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
-    }),
+    formState: OnyxEntry<Form>;
 
     /** Session details for the user */
-    session: PropTypes.shape({
-        /** accountID of current user */
-        accountID: PropTypes.number,
-    }),
+    session: OnyxEntry<Session>;
 
     /** policyID for main workspace */
-    activePolicyID: PropTypes.string,
-};
-const defaultProps = {
-    reports: {},
-    policies: {},
-    formState: {
-        isLoading: false,
-        errorFields: {},
-    },
-    session: {
-        accountID: 0,
-    },
-    activePolicyID: null,
+    activePolicyID: OnyxEntry<Required<Account>['activePolicyID']>;
 };
 
-function WorkspaceNewRoomPage(props) {
+type WorkspaceNewRoomPageProps = WorkspaceNewRoomPageOnyxProps;
+
+function WorkspaceNewRoomPage({policies, reports, formState, session, activePolicyID}: WorkspaceNewRoomPageProps) {
     const styles = useThemeStyles();
+    const isFocused = useIsFocused();
     const {translate} = useLocalize();
     const {isOffline} = useNetwork();
     const {isSmallScreenWidth} = useWindowDimensions();
-    const [visibility, setVisibility] = useState(CONST.REPORT.VISIBILITY.RESTRICTED);
-    const [policyID, setPolicyID] = useState(props.activePolicyID);
-    const [writeCapability, setWriteCapability] = useState(CONST.REPORT.WRITE_CAPABILITIES.ALL);
-    const wasLoading = usePrevious(props.formState.isLoading);
+    const [visibility, setVisibility] = useState<FormValues['visibility']>(CONST.REPORT.VISIBILITY.RESTRICTED);
+    const [policyID, setPolicyID] = useState<FormValues['policyID']>(activePolicyID);
+    const [writeCapability, setWriteCapability] = useState<FormValues['writeCapability']>(CONST.REPORT.WRITE_CAPABILITIES.ALL);
+    const wasLoading = usePrevious(!!formState?.isLoading);
     const visibilityDescription = useMemo(() => translate(`newRoomPage.${visibility}Description`), [translate, visibility]);
     const isPolicyAdmin = useMemo(() => {
         if (!policyID) {
             return false;
         }
 
-        return ReportUtils.isPolicyAdmin(policyID, props.policies);
-    }, [policyID, props.policies]);
-    const [newRoomReportID, setNewRoomReportID] = useState(undefined);
+        return ReportUtils.isPolicyAdmin(policyID, policies);
+    }, [policyID, policies]);
+    const [newRoomReportID, setNewRoomReportID] = useState<string>();
 
     /**
-     * @param {Object} values - form input values passed by the Form component
+     * @param values - form input values passed by the Form component
      */
-    const submit = (values) => {
-        const participants = [props.session.accountID];
+    const submit = (values: FormValues) => {
+        const participants = session?.accountID ? [session.accountID] : [];
```

Comment:
This will change original logic if `session.accountID` defaults to 0. (`[0]` -> `[]`)
Need to test if this doesn't cause any unexpected behavior.

Reply:
Please also test this flow. Not specified in Tests Steps

Reply:
@situchan Thanks for the heads up! Feedback applied 😄

Reply:
@situchan Creating a new room seems to be working correctly from my side 😄